spec/controllers/service_controller_spec.rb

```diff
@@ -165,17 +165,37 @@
       vm
     end
 
-    it 'returns VMs associated to selected Service' do
-      report_data_request(
-        :model         => 'Vm',
-        :parent_model  => 'Service',
-        :parent_id     => service.id,
-        :active_tree   => 'svcs_tree',
-        :parent_method => 'all_vms'
-      )
-      results = assert_report_data_response
-      expect(results['data']['rows'].length).to eq(1)
-      expect(results['data']['rows'][0]['long_id']).to eq(vm.id)
+    describe "#report_data" do
+      it 'returns VMs associated to the selected Service' do
+        report_data_request(
+          :model         => 'Vm',
+          :parent_model  => 'Service',
+          :parent_id     => service.id,
+          :active_tree   => 'svcs_tree',
+          :parent_method => 'all_vms'
+        )
+        results = assert_report_data_response
+        expect(results['data']['rows'].length).to eq(1)
+        expect(results['data']['rows'][0]['long_id']).to eq(vm.id)
+      end
+    end
+
+    # Request URL:http://localhost:3000/service/tree_select?id=s-10r219
+    describe '#tree_select' do
+      render_views
+
+      it 'renders GTL of VMs associated to the selected Service' do
```

Comment:
can you add more examples on how to test this further?

Reply:
I want to add real testcases. One is comming here: https://github.com/ManageIQ/manageiq-ui-classic/pull/2751

Reply:
And once we agree on this there's going to be a new article in https://github.com/ManageIQ/guides

Reply:
C-n-ped another example into the description of this PR.